lib/store/src/csf/testing-utils/index.ts

```diff
@@ -0,0 +1,117 @@
+import {
+  isExportStory,
+  AnyFramework,
+  AnnotatedStoryFn,
+  StoryAnnotations,
+  ComponentAnnotations,
+  ProjectAnnotations,
+  Args,
+  StoryContext,
+} from '@storybook/csf';
+
+import { prepareStory } from '../prepareStory';
+import { normalizeStory } from '../normalizeStory';
+import { HooksContext } from '../../hooks';
+import { normalizeComponentAnnotations } from '../normalizeComponentAnnotations';
+import { getValuesFromArgTypes, normalizeProjectAnnotations } from '..';
+import type { CSFExports, TestingStoryPlayFn, TestingStory } from './types';
+
+export * from './types';
+
+if (process.env.NODE_ENV === 'test') {
+  // eslint-disable-next-line global-require
+  const { default: addons, mockChannel } = require('@storybook/addons');
+  addons.setChannel(mockChannel());
+}
+
+let GLOBAL_STORYBOOK_CONFIG = {};
+
+export function setGlobalConfig<TFramework extends AnyFramework = AnyFramework>(
+  config: ProjectAnnotations<TFramework>
+) {
+  GLOBAL_STORYBOOK_CONFIG = config;
+}
+
+export function composeStory<
+  TFramework extends AnyFramework = AnyFramework,
+  TArgs extends Args = Args
+>(
+  story: AnnotatedStoryFn<TFramework, TArgs> | StoryAnnotations<TFramework, TArgs>,
+  meta: ComponentAnnotations<TFramework, TArgs>,
+  globalConfig: ProjectAnnotations<TFramework> = GLOBAL_STORYBOOK_CONFIG,
+  defaultConfig: ProjectAnnotations<TFramework> = {}
+) {
+  if (story === undefined) {
+    throw new Error('Expected a story but received undefined.');
+  }
+
+  const projectAnnotations = { ...defaultConfig, ...globalConfig };
+
+  const normalizedMeta = normalizeComponentAnnotations(meta);
+
+  const normalizedStory = normalizeStory(story.name, story, normalizedMeta);
+
+  const normalizedProjectAnnotations = normalizeProjectAnnotations(projectAnnotations);
+
+  const preparedStory = prepareStory<TFramework>(
+    normalizedStory,
+    normalizedMeta,
+    normalizedProjectAnnotations
+  );
+
+  const defaultGlobals = getValuesFromArgTypes(globalConfig.globalTypes);
+
+  const composedStory = (extraArgs: Partial<TArgs>) => {
+    const context: Partial<StoryContext> = {
+      ...preparedStory,
+      hooks: new HooksContext(),
+      globals: defaultGlobals,
+      args: { ...preparedStory.initialArgs, ...extraArgs },
+    };
+
+    return preparedStory.unboundStoryFn(context as StoryContext);
+  };
+
+  composedStory.storyName = story.storyName || story.name;
```

Comment:
Did we lose the bit where we were going to pull this off the normalized (or prepared) story?

Reply:
No, just not pushed!